### ISSUE TITLE
Edge case in WordPress.WP.I18n.MissingTranslatorsComment sniff

### ISSUE THREAD

**Issue by dingo-d** [WordPress/WordPress-Coding-Standards#1980]
## Bug Description

Just got a weird edge case triggering the `WordPress.WP.I18n.MissingTranslatorsComment` sniff, probably because of the combination of `%c` code in my translatable string (not using `sprintf` like functions).

## Minimal Code Snippet

```php
<?php

Components::render('paragraph', [ // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
	'paragraphContent' => esc_html__(
		'{%conference%} - this will denote the name of the conference.',
		'text-domain'
	),
	'componentClass' => 'description'
]),
```

The text is used as a description.

## Error Code
```bash
 78 | WARNING | A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
    |         | (WordPress.WP.I18n.MissingTranslatorsComment)

```
## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.4.14
| PHP_CodeSniffer version | 3.5.8
| WPCS version            | 2.3.0
| WPCS install type       | composer

**Comment by jrfnl** [comment]
`%c` is a [valid placeholder](https://www.php.net/manual/en/function.sprintf.php) for `sprintf()`-like functions, which is why the sniff _will_ trigger on it.

As translations will normally be done _before_ placeholder replacement (to keep the (amount of) translatable strings manageable) and the `sprintf()` can still be done in another part of the code base (doesn't have to be within the same function/file), I wouldn't consider this a false positive.

If the string is passed to `sprintf()` as some point before it is rendered, the literal `%`, should be written as `%%`.
If it is _not_ passed to `sprintf()` along the way, IMO it is bad practice to use literal `%`'s within translatable strings as it is just as confusing for translators as it is for static analysis tools like this.

**Comment by dingo-d** [comment]
Hmmm, I never thought that I could pass this to `sprintf()` at some point (because in my codebase I don't). That's a case I really didn't think would occur. It kinda seems a bit code smelly to me tbh (getting a translatable string with a placeholder from somewhere in the `sprintf()` function).

Thanks for clearing it up. I'll just use something else as a generic placeholder identifier, or put a space after it. 👍🏼

**Comment by jrfnl** [comment]
> It kinda seems a bit code smelly to me tbh (getting a translatable string with a placeholder from somewhere in the sprintf() function).

Just for my perception: what do you mean by this ?

To illustrate what I meant, take [this example from TGMPA](https://github.com/TGMPA/TGM-Plugin-Activation/blob/2d34264f4fdcfcc60261d490ff2e689f0c33730c/class-tgm-plugin-activation.php#L338-L411) (note: same principle is used elsewhere too, as well as in Core). Here the translated strings are retrieved early and allowed to be filtered by the implementing application. Only when they are actually used, the strings are subsequently passed to `sprintf()`, escaped and send to screen.

**Comment by dingo-d** [comment]
I stand corrected, as usual 😅
